modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,264 @@
+      ext = f.split(':')[1]
+      # Split string into name and extension and check status
+      status = get_incomplete_status(url, "*" , "1" , ext)
+      next unless status == 404
+      next unless ext.size <= 3
+
+      @charset_duplicates.each do |x|
+        if get_complete_status(url, x , ext) == 404
+          @files << "#{url}*~#{x}.#{ext}*"
+        end
+      end
+
+      if ext.size < 3
+        for c in @charset_extensions
+          @queue_ext << (f + c )
+        end
+      end
+    end
+  end
+
+  def scan
+    while !@queue.empty?
+      url = @queue.pop
+      status = get_status(url , "1" , "*")
+      # Check strings only upto 6 chars in length
+      next unless status == 404
+      if url.size == @name_size
+        @charset_duplicates.each do |x|
+          if get_status(url , x , "") == 404
+            @dirs << "#{url}*~#{x}"
+          end
+        end
+        # If a url exists then add to new queue for extension scan
+        for ext in @charset_extensions
+          @queue_ext << ( url + ':' + ext )
+          @threads << framework.threads.spawn("scanner", false) { scanner }
+        end
+      else
+        @charset_duplicates.each do |x|
+          if get_complete_status(url, x , "") == 404
+            @dirs << "#{url}*~#{x}"
+            break
+          end
+        end
+        if get_incomplete_status(url, "" , "1" , "") == 404
+          for ext in @charset_extensions
+            @queue_ext << ( url + ':' + ext )
+            @threads << framework.threads.spawn("scanner", false) { scanner }
+          end
+        elsif url.size < @name_size
+          for c in @charset_names
+            @queue  <<(url +c)
+          end
+        end
+      end
+    end
+  end
+
+  def reduce
+    # Reduce the total charset for filenames by checking if a character exists in any of the files
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*#{c}*~1*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @charset_names << c
+      end
+    end
+  end
+
+  def ext
+    # Reduce the total charset for extensions by checking if a character exists in any of the extensions
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*~1.*#{c}*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @charset_extensions << c
+      end
+    end
+  end
+
+  def dup
+    # Reduce the total charset for duplicate files/folders
+    array = [*('1'..'9')]
+    array.each do |c|
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*~#{c}.*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @charset_duplicates << c
+      end
+    end
+  end
+
+  def run
+    unless is_vul
+      print_status("Target is not vulnerable, or no shortname scannable files are present.")
+      return
+    end
+    if datastore['PATH'][-1] != '/'
```

Comment:
`end_with?` is preferred.

```suggestion
    unless datastore['PATH'].end_with? '/'
```

Additionally, `datastore` properties should not be modified within the module.

Reply:
I think this is better now but if indentation is still messed up we'll look at this again